src/Workspaces/Core/Portable/Workspace/Host/TextFactory/ITextFactoryService.cs

```diff
@@ -28,5 +28,15 @@ internal interface ITextFactoryService : IWorkspaceService
         /// </exception>
         /// <exception cref="IOException">An IO error occurred while reading from the stream.</exception>
         SourceText CreateText(Stream stream, Encoding defaultEncoding, CancellationToken cancellationToken = default(CancellationToken));
+
+
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from a reader.
+        /// </summary>
+        /// <param name="reader">The <see cref="TextReader"/> to read the text from.</param>
+        /// <param name="encoding">Specifies an encoding for the <see cref="SourceText"/>SourceText. it could be even null</param>
```

Comment:
~~it could be even null~~

"If null, the encoding will be determined based on the contents of the reader."